pandas/core/indexes/base.py

```diff
@@ -2545,7 +2560,15 @@ def inferred_type(self) -> str_t:
         """
         Return a string of the type inferred from the values.
         """
-        return lib.infer_dtype(self._values, skipna=False)
+        try:  # fastpath numeric indexes
+            return {
+                "i": "integer",
+                "u": "integer",
+                "f": "floating",
+                "c": "complex",
+            }[self.dtype.kind]
```

Comment:
this doesnt change anything for our nullable dtypes, but in principle could be an API change for 3rd party EAs.  worth caring about @mroeschke ?

Reply:
I think this would be a good API change for `ArrowDtype`, but not sure what else subsequently relies on the output of `Index.inferred_type`

Reply:
This change I proposed isn't strictly needed, I just added it as an optimization. I could also remove it here and try do add something similar to `lib.infer_dtype` instead. It's also a better location for this IMO and at least we can discuss if there's some down side to doing this seperately from the `NumericIndex` removal process.

Reply:
to avoid the api change you can do:

```
if isinstance(self.dtype, np.dtype) and self.dtype.kind in "iufc":
    [...]
return lib.infer_dtype(...)
```

Reply:
Ok, I can do it like that.